[Empty States] Add empty state for search errors

bjtitus opened the pull request:
| 📘 Part of: #3102 |
|:---:|

* Adds empty state for the Podcast section of search.
* Adds a separate failed state for any errors during search. This is to differentiate with an empty search result.
* Adds a Try Again button to retry the search calls

| Empty | Both Failed | One Fails |
|--|--|--|
| ![Simulator Screenshot - iPhone 16 - 2025-05-26 at 11 26 37](https://github.com/user-attachments/assets/021dccea-8607-4876-8df9-44b84d22ed08) | ![Simulator Screenshot - iPhone 16 - 2025-05-26 at 22 01 00](https://github.com/user-attachments/assets/4f9d28ce-6081-489d-ad5a-490c2e46638a) | ![Simulator Screenshot - iPhone 16 - 2025-05-26 at 22 00 29](https://github.com/user-attachments/assets/713346e4-e344-4c7f-a963-b27535590bb3) |

## To test

* Load [these scripts](https://github.com/user-attachments/files/20444449/search_state_proxyman_scripts.config.zip) into Proxyman from Tools > Import Settings > From Proxyman > Scripting
* Enable "Empty Podcast Search" and run a search
* ✅ Verify that the new Empty state appears
* Disable the "Empty Podcast Search" and Enable the "Failed Search"
* Run a new search
* ✅ Verify that the new Error states appear


## Checklist
- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

bjtitus committed:
Add empty state for search errors

bjtitus committed:
Add empty state for podcasts search

bjtitus committed:
Add error handling for podcasts search

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-ios/milestone/91">7.90 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

bjtitus committed:
Remove unused errors

bjtitus committed:
Add Try Again button to search failure states

bjtitus committed:
Unify Empty search state if both search calls fail

bjtitus committed:
Use template image for search results

bjtitus committed:
Change button style for Search Results

bjtitus committed:
Fix icon and link text coloring

danielebogo reviewed (changes requested):
@bjtitus overall it looks good but I found a background color problem

| 1 | 2 |
| -------- | ------- |
| ![IMG_0160](https://github.com/user-attachments/assets/f80a6021-3c7f-4ea3-8855-ef2eb0a1e432) | ![IMG_0161](https://github.com/user-attachments/assets/4661205e-57a0-4a6a-8648-fdebdb7c96ee) |

bjtitus committed:
Expand HStack to fill search results view

bjtitus committed:
Set background for empty search results view container

bjtitus commented:
@danielebogo Ugh, sorry. This broke when I added the centering. Fixed now!

| | |
|--|--|
| ![Simulator Screenshot - iPhone 16 - 2025-05-27 at 09 00 24](https://github.com/user-attachments/assets/f4fc8ca0-c87f-4372-9c03-89beea9add3b) | ![Simulator Screenshot - iPhone 16 - 2025-05-27 at 09 00 28](https://github.com/user-attachments/assets/4153f32d-ae7e-4e5b-8fa3-98b80cdd15e6) |

danielebogo reviewed (approved):
Thanks for the quick fix! :shipit:
